### PR TITLE
Cc fix

### PR DESCRIPTION
**Pull request by hunharibo**
Please review.
Generic character creation scren and options. Offloaded to XML instead of being hardcoded like TW did it.

### CODE REVIEW

**Review by Z3rca — APPROVED** [review]
I was at first confused about the weapons.xml change but, I think it just enhance the readability. The rest seems to work like I orginally intended, with customizability of XML files. thats great.

In General, I would suggest, if possible to make smaller commits, but that is just a personal preference.

**Comment by hunharibo** [comment]
Yeah sorry, its not a really clean commit :( I left some experimentation in it. Will try to pay more attention in the future.

File: HarmonyPatches/CharacterCreationPatches.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(CampaignGameManager), "LaunchSandboxCharacterCreation")]
+        public static bool Prefix2(CampaignGameManager __instance)
+        {
+            CharacterCreationState gameState = Game.Current.GameStateManager.CreateState<CharacterCreationState>(new object[]
+{
+                new TOWCharacterCreationContent()
+});
```
Comment (tristanka) [inline HarmonyPatches/CharacterCreationPatches.cs]: Indentation is a little messed up here

File: CharacterCreation/TOWCharacterCreationContent.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                            category = stage3Menu.AddMenuCategory(delegate ()
+                            {
+                                return GetSelectedCulture().StringId == culture;
+                            });
+                            break;
+                        default:
+                            break;
+                    }
+                    
+                    var relevantOptions = _options.FindAll(x => x.StageNumber == i && x.Culture == culture);
```
Comment (tristanka) [inline CharacterCreation/TOWCharacterCreationContent.cs]: `x.Culture == culture` -> `x.Culture.Equals(culture)` just to ensure value equality and avoid any weird reference inequalities in the future? Probably wouldn't happen but we really just care about value equality

**Review by tristanka — COMMENTED** [review]
Also, didn't really look into why this is happening, but the culture list only shows the two custom cultures for New Campaign as well, but when you move forward, it uses the base TaleWorlds character creation content. I don't think this is a big issue right now, just wanted to point it out in case you hadn't noticed.

**Review by tristanka — APPROVED** [review]
🎉

**Pull request merged**